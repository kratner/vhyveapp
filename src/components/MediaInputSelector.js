import React from "react";
import styled from "styled-components";
const MediaSelectorContainer = styled.div`
  font-size: 0.5em;
`;
const MicSelectorContainer = styled(MediaSelectorContainer)``;
const CameraSelectorContainer = styled(MediaSelectorContainer)``;
const InputSelectorContainer = styled.div``;
const MediaInputItem = styled.div`
  height: 2.5em;
  display: flex;
  align-items: center;
  cursor: pointer;
  transition: 0.25s;
  padding-left: 0.5em;
  &:hover {
    background-color: #ffffff25;
  }
  &.selected {
    background-color: #ffffff15;
  }
`;
const MediaInputTypeLabel = styled.div`
  font-size: 0.8em;
  font-weight: bold;
  padding: 0.25em 0.15em;
`;
const MProperty = styled.span``;
const MLabel = styled(MProperty)``;
const MediaInputSelector = props => {
  const handleAudioInputItemClick = (evt, device) => {
    const audiodeviceselectors = document.getElementsByClassName(
      "audiodeviceselector"
    );
    for (let i = 0; i < audiodeviceselectors.length; i++) {
      audiodeviceselectors[i].classList.remove("selected");
    }
    evt.currentTarget.classList.add("selected");
    props.onAudioItemClick(device);
  };
  const handleVideoInputItemClick = (evt, device) => {
    const videodeviceselectors = document.getElementsByClassName(
      "videodeviceselector"
    );
    for (let i = 0; i < videodeviceselectors.length; i++) {
      videodeviceselectors[i].classList.remove("selected");
    }
    evt.currentTarget.classList.add("selected");
    props.onVideoItemClick(device);
  };
  const AudioInputDeviceList = props.audioInputDevices.map((node, index) => {
    const cls = (index === 0 ? "selected" : "") + ` audiodeviceselector`;
    return (
      <MediaInputItem
        className={cls}
        key={index}
        onClick={evt => {
          handleAudioInputItemClick(evt, node);
        }}
      >
        <MLabel>{node.label}</MLabel>
      </MediaInputItem>
    );
  });
  const VideoInputDeviceList = props.videoDevices.map((node, index) => {
    const cls = (index === 0 ? "selected" : "") + ` videodeviceselector`;
    return (
      <MediaInputItem
        className={cls}
        key={index}
        onClick={evt => {
          handleVideoInputItemClick(evt, node);
        }}
      >
        <MLabel>{node.label}</MLabel>
      </MediaInputItem>
    );
  });
  return (
    <InputSelectorContainer>
      <MediaInputTypeLabel>Select Audio Device</MediaInputTypeLabel>
      <MicSelectorContainer id="micselectorcontainer">
        {AudioInputDeviceList}
      </MicSelectorContainer>
      <MediaInputTypeLabel>Select Camera</MediaInputTypeLabel>
      <CameraSelectorContainer id="cameraselectorcontainer">
        {VideoInputDeviceList}
      </CameraSelectorContainer>
    </InputSelectorContainer>
  );
};

export default MediaInputSelector;
