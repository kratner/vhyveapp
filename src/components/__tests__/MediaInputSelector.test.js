import React from "react";
import { shallow } from "enzyme";
import MediaInputSelector from "../MediaInputSelector";

const audioInputDevices = [
  {
    deviceId: "default",
    kind: "audioinput",
    label: "Default - Internal Microphone (Built-in)",
    groupId: "0a683d153b8655aa5a09fc151c0f921c5a7050467a06ecd68f47d82746274b8f"
  },
  {
    deviceId:
      "e6b950fbc4fe58719ae20d9b18a0c3089b3895a0f3afca30c01b79e1ca295a44",
    kind: "audioinput",
    label: "Internal Microphone (Built-in)",
    groupId: "0a683d153b8655aa5a09fc151c0f921c5a7050467a06ecd68f47d82746274b8f"
  }
];

const audioOutputDevices = [
  {
    deviceId: "default",
    kind: "audiooutput",
    label: "Default - Internal Speakers (Built-in)",
    groupId: "0a683d153b8655aa5a09fc151c0f921c5a7050467a06ecd68f47d82746274b8f"
  },
  {
    deviceId:
      "ab7661d1168c9ae6e40d8df20dd26bf20ad4ca3f786a415943dcdf5b1d03b084",
    kind: "audiooutput",
    label: "Internal Speakers (Built-in)",
    groupId: "0a683d153b8655aa5a09fc151c0f921c5a7050467a06ecd68f47d82746274b8f"
  }
];
const videoDevices = [
  {
    deviceId:
      "ad73df458d8a30aed8c537eec886310f70deab646e4046d04f987960163ec3bc",
    kind: "videoinput",
    label: "FaceTime HD Camera (Built-in) (05ac:8510)",
    groupId: "28514e297a1ffe291d3174d4d226ea07a6ea13a35005ff0c8474712c00ff184f"
  }
];
describe("MediaInputSelector", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <MediaInputSelector
        audioInputDevices={audioInputDevices}
        audioOutputDevices={audioOutputDevices}
        videoDevices={videoDevices}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
