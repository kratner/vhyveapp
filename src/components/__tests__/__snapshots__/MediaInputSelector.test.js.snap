// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaInputSelector renders correctly 1`] = `ShallowWrapper {}`;
